src/autoscaler/healthendpoint/health_readiness_test.go

```diff
@@ -318,4 +322,55 @@ var _ = Describe("Health Readiness", func() {
 
 	})
 
+	Context("pprof endpoint", func() {
+		When("basic auth is not configured", func() {
+			BeforeEach(func() {
+				config.HealthCheckUsername = ""
+				config.HealthCheckPassword = ""
+			})
+			It("should not be available", func() {
+				apitest.New().
+					Handler(healthRoute).
+					Get("/debug/pprof").
+					Expect(t).
+					Assert(bodyContainsSubstring(false, "Types of profiles available")).
+					Status(http.StatusOK).
+					End()
+			})
+		})
+
+		When("basic auth is configured", func() {
+			It("should be available", func() {
+				apitest.New().
+					Handler(healthRoute).
+					Get("/debug/pprof").
+					BasicAuth("test-user-name", "test-user-password").
+					Expect(t).
+					Assert(bodyContainsSubstring(true, "Types of profiles available")).
+					Status(http.StatusOK).
+					End()
+			})
+		})
+	})
 })
+
+func bodyContainsSubstring(doesContain bool, substr string) func(res *http.Response, _ *http.Request) error {
+	return func(res *http.Response, _ *http.Request) error {
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			return errors.New("failed reading body")
+		}
+		if Expect(string(b)).To(Invert(!doesContain, ContainSubstring(substr))) {
```

Comment:
you can use ToNot instead of having invert

Reply:
yeah, I tackled from that wrong side

Reply:
should be better now.